packages/core/src/parser/util.ts

```diff
@@ -506,6 +506,63 @@ export function stopBefore<N extends Returnable>(
 	}
 }
 
+/**
+ * @returns A parser that is based on the passed-in `parser`, but concatenates lines
+ * together when we reach, in order:
+ * - a backslash
+ * - whitespace (optional)
+ * - a newline
+ * - whitespace (optional)
+ */
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: InfallibleParser<N>,
+): InfallibleParser<N>
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: Parser<N>,
+): Parser<N>
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: Parser<N>,
+): Parser<N> {
+	return (src, ctx): Result<N> => {
+		let wrappedStr = src.sliceToCursor(0)
+		const wrappedSrcCursor = wrappedStr.length
+		const indexMap: IndexMap = []
+
+		while (src.canRead()) {
+			wrappedStr += src.readUntil('\\')
+			if (!src.canRead()) {
+				break
```

Comment:
When this happens, Minecraft throws an error `Line continuation at end of file`, so we should probably show an error in vscode as well.

Reply:
What happens when there is a \ at the end of the line followed by an empty line?

Reply:
> When this happens, Minecraft throws an error `Line continuation at end of file`, so we should probably show an error in vscode as well.

confirmed this in-game. implementing fix

---

> What happens when there is a \ at the end of the line followed by an empty line (or an empty line before EOF)?

looks like it's fine as long as its not right before EOF

the following `reload`s fine in-game and outputs as the following:

![image](https://github.com/SpyglassMC/Spyglass/assets/13565346/e96eb610-dfdf-4e96-922b-7e6bf2be34c7)

![image](https://github.com/SpyglassMC/Spyglass/assets/13565346/6d443366-ea5e-4cc7-9af7-69d9edc8b481)

Reply:
[`0548d94` (#1180)](https://github.com/SpyglassMC/Spyglass/pull/1180/commits/0548d94fcdf84656a3532de7bd3d8afdde983a8f)